src/Avalonia.Themes.Fluent/Controls/ColorSpectrum.xaml

```diff
@@ -0,0 +1,138 @@
+                         RadiusY="{Binding CornerRadius, RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource BottomRightCornerRadiusDoubleValueConverter}}" />-->
+              <Rectangle x:Name="PART_SpectrumOverlayRectangle"
+                         IsHitTestVisible="False"
+                         HorizontalAlignment="Stretch"
+                         VerticalAlignment="Stretch"
+                         IsVisible="{TemplateBinding Shape, Converter={StaticResource EnumValueEquals}, ConverterParameter='Box'}" />
+                         <!--RadiusX="{Binding CornerRadius, RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource TopLeftCornerRadiusDoubleValueConverter}}"
+                         RadiusY="{Binding CornerRadius, RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource BottomRightCornerRadiusDoubleValueConverter}}" />-->
+              <Ellipse x:Name="PART_SpectrumEllipse"
+                       IsHitTestVisible="False"
+                       HorizontalAlignment="Stretch"
+                       VerticalAlignment="Stretch"
+                       IsVisible="{TemplateBinding Shape, Converter={StaticResource EnumValueEquals}, ConverterParameter='Ring'}" />
+              <Ellipse x:Name="PART_SpectrumOverlayEllipse"
+                       IsHitTestVisible="False"
+                       HorizontalAlignment="Stretch"
+                       VerticalAlignment="Stretch"
+                       IsVisible="{TemplateBinding Shape, Converter={StaticResource EnumValueEquals}, ConverterParameter='Ring'}" />
+              <Canvas x:Name="PART_InputTarget"
+                      Background="Transparent"
+                      HorizontalAlignment="Stretch"
+                      VerticalAlignment="Stretch">
+                <Grid x:Name="PART_SelectionEllipsePanel">
+                  <Ellipse x:Name="FocusEllipse"
+                           Margin="-2"
+                           StrokeThickness="2"
+                           IsHitTestVisible="False"
+                           HorizontalAlignment="Stretch"
+                           VerticalAlignment="Stretch" />
+                  <Ellipse x:Name="SelectionEllipse"
+                           StrokeThickness="2"
+                           IsHitTestVisible="False"
+                           HorizontalAlignment="Stretch"
+                           VerticalAlignment="Stretch"
+                           ToolTip.VerticalOffset="-20"
+                           ToolTip.Placement="Top">
+                    <ToolTip.Tip>
+                      <ToolTip x:Name="PART_ColorNameToolTip" />
+                    </ToolTip.Tip>
+                  </Ellipse>
+                </Grid>
+              </Canvas>
+              <Rectangle x:Name="RectangleBorder"
```

Comment:
Should it be replaced with Border? To have radiusx/y props.

Reply:
I did discover https://github.com/AvaloniaUI/Avalonia/issues/4066 here when I commented out the RadiusX/Y setters. I know Border is a work-around but honestly I would rather add these properties to rectangle. Border seems overkill here and it should be possible to use shape primitives.

So bottom line, I would like to actually add RadiusX/Y to rectangle and then re-enable the commented code here.

Reply:
I think it's fine to keep this open when the PR is merged and then address it later.

Reply:
The PR #7914 adding corner radius to Rectangle was committed, by you -- thanks! So I went ahead and updated this PR. I had to add a new converter (and make a small breaking change to an existing converter) but ColorSpectrum now supports rounded corners.

Marking this as resolved.